imod/mf6/hfb.py

```diff
@@ -281,11 +282,28 @@ def __init__(
         super().__init__(locals())
         self.dataset["print_input"] = print_input
 
-        self.dataset = self.dataset.merge(geometry.to_xarray())
+        self.line_data = geometry
+
+    def _get_variable_names_for_gdf(self) -> list[str]:
+        return [
+            self._get_variable_name(),
+            "geometry",
+        ] + self._get_vertical_variables()
 
     @property
-    def geometry(self) -> np.ndarray[object]:
-        return self.dataset["geometry"].values
+    def line_data(self) -> gpd.GeoDataFrame:
+        variables_for_gdf = self._get_variable_names_for_gdf()
+        return gpd.GeoDataFrame(
+            self.dataset[variables_for_gdf].to_dataframe(),
+            geometry="geometry",
```

Comment:
I don't think we need the `geometry="geometry" argument. I think because there is already a variable called geometry in the dataset that will automatically be used as the geometry column.

Reply:
I think it is better to specify that explitictly to geopandas, as the geometry column is crucial to its functioning. On one hand it might be that the argument name "geometry" changes, on the other hand it might be that behavior changes for inferring this "geometry" column if no geometry is explicitly provided. I prefer the first case, as it provides a clearer erro message.